Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApayaTokenizationViewModel.swift

```diff
@@ -217,21 +222,12 @@ class ApayaTokenizationViewModel: PaymentMethodTokenizationViewModel, ExternalPa
         }
         .done { paymentMethod in
             DispatchQueue.main.async {
-                PrimerDelegateProxy.onTokenizeSuccess(paymentMethod, resumeHandler: self)
-                PrimerDelegateProxy.onTokenizeSuccess(paymentMethod, { err in
-                    if let err = err {
-                        self.handleFailedTokenizationFlow(error: err)
-                    } else {
-                        self.handleSuccessfulTokenizationFlow()
-                    }
-                })
+                self.handleContinuePaymentFlowWithPaymentMethod(paymentMethod)
             }
         }
-        .catch { err in
+        .catch { error in
             DispatchQueue.main.async {
```

Comment:
I think it's safe to start removing switching to main queue from a lot of places and just put it within the `PrimerDelegateProxy` and the handlers that push screens.

Reply:
Addressed ✅